cpp/betweenness_centrality_module/algorithm/betweenness_centrality.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <stack>
+#include <vector>
+
+#include <mg_graph.hpp>
+
+namespace betweenness_centrality_util {
+
+///
+///@brief A method that performs an augmented BFS algorithm to count the number of shortest paths
+/// between the source node and all nodes in the given graph. This method has O(m) time complexity.
+///
+///@param source_node The starting node of the bfs traversal
+///@param graph The graph that is traversed by the BFS algorithm
+///@param visited The container for storing nodes in the order in which they are discovered by the BFS algorithm
+///@param predecessors The container for storing nodes that are located just before a specific node on
+/// the shortest path from the source node to the beforementioned node
+///@param shortest_paths_counter The container for storing the number of shortest paths from the source node to
+/// all nodes in the given graph
+///
+void BFS(const std::uint64_t source_node, const mg_graph::GraphView<> &graph, std::stack<std::uint64_t> &visited,
+         std::vector<std::vector<std::uint64_t>> &predecessors, std::vector<std::uint64_t> &shortest_paths_counter);
+
+///
+///@brief An in-place method that normalizes a vector by multiplying each component by a given constant.
+///
+///@param vec The vector that should be normalized
+///@param constant The constant with which the components of a vector are multiplied
+///
+void Normalize(std::vector<double> &vec, double constant);
+
+}  // namespace betweenness_centrality_util
```

Comment:
This can be in anonymous namespace in `betweenness_centrality.cpp`. No need to define it in header.

Reply:
How to include the BFS declaration in the betweenness_centrality_test then?

Reply:
That's true, you can leave it here then, but I would still move `Normalize` (if possible :smile: )